lib/resolve-value.js

```diff
@@ -27,26 +29,61 @@ function toString(value) {
 var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal debugging*/_debugIsInternal) {
 	var debugIndent = _debugIsInternal ? '\t' : '';
 
+	var matchingVarDecl = undefined;
+	var RE_VAR_FUNC_G = new RegExp(RE_VAR_FUNC.source, 'g');
 	var resultantValue = toString(decl.value);
 	var warnings = [];
-
-	var variablesUsedInValueMap = {};
-	// Use `replace` as a loop to go over all occurrences with the `g` flag
-	resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		variablesUsedInValueMap[variableName] = true;
-	});
-	var variablesUsedInValue = Object.keys(variablesUsedInValueMap);
+	
+	// Match all variables first so we can later on if there are circular dependencies
+	var variablesUsedInValue = [];
+	// Create a temporary variable, storing resultantValue variable value
+	var resultantValueTemp = resultantValue;
```

Comment:
`resultantValueTemp` -> `remainingVariableValue` to be a bit more descriptive

Since we keep clearing it out as go